fix: sdk generator - date support and doc (#529)
AmadeusITGroup/otter: packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/readme.md

Reviewer: This documentation is provided only to new SDK, it means that the solution documented is not provided/accessible to previously generated owner.
Maybe it should also be provided to the Otter repository documentation (in `docs/` or on the `@ama-sdk/core` readme)

Author: For the sdk-core doc, they are a few things that needs to be updated, I will tackle it a dedicated PR. But yes that's a good point.

yarn test
```

## Manage dates

### The timezone issue

> Managing dates with timezones have always been a bit painful in front end applications.

Reviewer 2: ```suggestion
Managing dates with timezones has always been a bit painful in front end applications.
```

Let's give a concrete example to understand the problem:

> In the case of an API returning the date and hour of your flight : 2023-07-10T00:37:00.000+07:00

Reviewer 2: ```suggestion
An API returns the date and hour of your flight in the timezone of the airport location. In our use case, let's say the departure airport is on GTM+7 : 2023-07-10T00:37:00.000+07:00.
```

The timezone sent is the one from the airport, here GMT+7. If you just use the Date(), the computer browser will convert this in its own timezone.
You will end up having for example if the user is in GMT+2 : 2023-07-09T19:37:00.000+02:00.

> You don't want that, you want the exact date time of the flight, at the airport timezone, not the user one.

Reviewer 2: ```suggestion
You don't want that, you want the exact date time of the flight at the airport timezone, not the time of your user's computer when the flight will depart.
```

> However, sometimes you might need this timezone information, for example to be able to display to the user that the flight is in X hours (need both airport and user timezone).

Reviewer 2: ```suggestion
However, there might be cases where you might still need this timezone information. 
For example, you want to be able to display that the flight is in X hours. You will need to compute this information with the two timezones -- the airport's and the user's.
```

> However, sometimes you might need this timezone information, for example to be able to display to the user that the flight is in X hours (need both airport and user timezone).

Reviewer 3: ```suggestion
However, you may sometimes need this timezone information, for example to be able to indicate to the user that the flight is in X hours (both the airport and the user timezones are needed).
```

> ### Solution proposed: utils.DateTime

Reviewer: The title may not be matching the content of the doc.
Actually the main purpose of the document under this section is to provide an example to get the date considering timezone.
Maybe it would make sense to move this example in a new section or to change the title of this one?

Author: The purpose of this section is to get rid of the timezone, keeping the timezone if the next one, I'll update the titles to be more accurate

> That's why the default Date behavior is to get rid of the timezone, by replacing Date by utils.Date.

Reviewer 2: ```suggestion
As we need to get rid the timezones more often than not, this will be our default behaviour. This is why, we introduced the utils.Date object to replace Date in those use cases.
```

Reviewer 3: Date or DateTime ?

import type { FlightCoreIfy } from './flight';

/**
 * @param baseRevive
 */
export function reviveFlightFactory<R extends typeof reviveFlight>(baseRevive: R) {
  const reviver = <T extends Flight = Flight>(data: any, dictionaries?: any) => {
    const originalData: any = {...data};
    const revivedData = baseRevive<FlightCoreIfy<T>>(data, dictionaries);
    if (!revivedData) {
      return;
    }
    revivedData.departureDateTimeConsideringTimezone = originalData.departureDateTimeConsideringTimezone && new Date(originalData.departureDateTimeConsideringTimezone)
      || originalData.departureDateTime && new Date(originalData.departureDateTime);
    return revivedData;
  };
  return reviver;
}
```

And export it here (src/models/core/flight/index.ts):
```typescript
export * from './flight';
export * from './flight.reviver';
```

And here (src/models/core/index.ts):
```typescript
export * from './flight/index';
```

You can now use departureDateTimeConsideringTimezone to access the timezone information.
See [utils.Date](https://github.com/AmadeusITGroup/otter/blob/main/packages/%40ama-sdk/core/src/fwk/date.ts) for more information.

> ### How to keep the timezone (not use utils.Datetime)

Reviewer 3: utils.Datetime or utils.Date ?
